mobile/ios/ios-application-manager.ts

```diff
@@ -0,0 +1,129 @@
+	private uninstallApplicationCallbackPtr: NodeBuffer = null;
+	
+	constructor(private device: Mobile.IiOSDevice,
+		private devicePointer: NodeBuffer,
+		private $childProcess: IChildProcess,
+		private $coreFoundation: Mobile.ICoreFoundation,
+		private $errors: IErrors,
+		private $injector: IInjector,
+		private $mobileDevice: Mobile.IMobileDevice,
+		private $logger: ILogger,
+		private $hostInfo: IHostInfo,
+		private $staticConfig: IStaticConfig) {
+			this.uninstallApplicationCallbackPtr = CoreTypes.am_device_mount_image_callback.toPointer(IOSApplicationManager.uninstallCallback); 
+		}
+		
+	private static uninstallCallback(dictionary: NodeBuffer, user: NodeBuffer): void { }
+		
+	public getInstalledApplications():  IFuture<string[]> {
+		return (() => {
+			return _(this.lookupApplications()).keys().sortBy((identifier: string) => identifier.toLowerCase()).value();
+		}).future<string[]>()();
+	}
+	
+	public installApplication(packageFilePath: string): IFuture<void> {
+		return (() => {
+			let installationProxy = this.$injector.resolve(iOSProxyServices.InstallationProxyClient, { device: this.device });
+			installationProxy.deployApplication(packageFilePath).wait();
+			installationProxy.closeSocket();
+		}).future<void>()();
+	}
+
+	public uninstallApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			let afc = this.device.startService(iOSProxyServices.MobileServices.INSTALLATION_PROXY);
+			try {
+				let result = this.$mobileDevice.deviceUninstallApplication(afc, this.$coreFoundation.createCFString(applicationId), null, this.uninstallApplicationCallbackPtr);
+				if(result !== 0) {
+					this.$errors.fail("AMDeviceUninstallApplication returned '%d'.", result);
+				}
+			} finally {
+			}
+
+			this.$logger.trace("Application %s has been uninstalled successfully.", applicationId);
+		}).future<void>()();
+	}	
+	
+	public startApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			if(this.$hostInfo.isWindows && (this.$staticConfig.enableDeviceRunCommandOnWindows === null || this.$staticConfig.enableDeviceRunCommandOnWindows === undefined)) {
+				this.$errors.fail("$%s device run command is not supported on Windows for iOS devices.", this.$staticConfig.CLIENT_NAME.toLowerCase());
+			}
+			
+			this.validateApplicationId(applicationId);
+			this.device.mountImage().wait();
+			
+			this.runApplicationCore(applicationId).wait();
+			this.$logger.info(`Successfully run application ${applicationId} on device with ID ${ this.device.deviceInfo.identifier}.`);
+		}).future<void>()();
+	}
+	
+	public stopApplication(applicationId: string): IFuture<void> {
+		let application = this.getApplicationById(applicationId);		
+		let gdbServer = this.createGdbServer();
+		return gdbServer.kill(application.CFBundleExecutable);
+	}
+	
+	public restartApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			this.$childProcess.exec(`${process.argv[0]} device stop ${applicationId}`).wait();
```

Comment:
`process.argv[0]` should be node, so you'll try executing `node device stop goshko.telerik.com`
Also spawning another process for this is not really expected for me, can't we just execute the logic behind `device stop` command instead of spawning? Also this code will fail in case we have local installation of CLI.